sycl/test-e2e/Matrix/SG32/joint_matrix_half.cpp

```diff
@@ -9,7 +9,6 @@
 // UNSUPPORTED: gpu-intel-dg2
 // REQUIRES: aspect-fp16
```

Comment:
aspect-fp16 does not guarantee that joint matrix support half.
We should add type based query to the hpp file as AMX of SPR does not support half 
Similar comment applies to the other cases that ise matrix-fp16

Reply:
I already did it.